pcdsdevices/fms.py

```diff
@@ -0,0 +1,259 @@
+CXI:SETRA:01:FLOW_RATE 60 monitor
+CXI:SETRA:01:SAMPLE_DURATION 60 monitor
+CXI:SETRA:01:CH0_SIZE 60 monitor
+CXI:SETRA:01:CH0_MASS 60 monitor
+CXI:SETRA:01:CH1_SIZE 60 monitor
+CXI:SETRA:01:CH1_MASS 60 monitor
+CXI:SETRA:01:CH2_SIZE 60 monitor
+CXI:SETRA:01:CH2_MASS 60 monitor
+CXI:SETRA:01:CH3_SIZE 60 monitor
+CXI:SETRA:01:CH3_MASS 60 monitor
+CXI:SETRA:01:CH4_SIZE 60 monitor
+CXI:SETRA:01:CH4_MASS 60 monitor
+CXI:SETRA:01:CH5_SIZE 60 monitor
+CXI:SETRA:01:CH5_MASS 60 monitor
+"""
+
+
+from ophyd import Component as Cpt, Device, EpicsSignalRO
+from ophyd import FormattedComponent as FCpt, EpicsSignalRO, Device
+
+class Setra5000(Device):
+    Timestamp = Cpt(EpicsSignalRO, ':TIME', kind='hinted')
+    Date = Cpt(EpicsSignalRO, ':DATE', kind='hinted')
+
+    SampleState = Cpt(EpicsSignalRO, ':SAMP_STATE', kind='hinted')
+    SampleDuration = Cpt(EpicsSignalRO, ':SAMPLE_TIME_RBV', kind='hinted')
+    SampleMode = Cpt(EpicsSignalRO, ':SAMPLE_MODE_RBV', kind='hinted')
+    MassMode = Cpt(EpicsSignalRO, ':MASS_MODE_RBV', kind='hinted')
+    
+    Temperature = Cpt(EpicsSignalRO, ':TEMP', kind='hinted')
+    RH = Cpt(EpicsSignalRO, ':RH', kind='hinted')
+    BarometricPressure = Cpt(EpicsSignalRO, ':BP', kind='hinted')
+    FlowRate = Cpt(EpicsSignalRO, ':FLOW_RATE', kind='hinted')
+    CO2 = Cpt(EpicsSignalRO, ':CO2', kind='hinted')
+    TVOC = Cpt(EpicsSignalRO, ':TVOC', kind='hinted')    
+
+    '''
+    ch0_size = Cpt(EpicsSignalRO, ':CH0_SIZE', kind='normal')
+    ch0_mass = Cpt(EpicsSignalRO, ':CH0_MASS', kind='normal')
+    ch1_size = Cpt(EpicsSignalRO, ':CH1_SIZE', kind='normal')
+    ch1_mass = Cpt(EpicsSignalRO, ':CH1_MASS', kind='normal')
+    ch2_size = Cpt(EpicsSignalRO, ':CH2_SIZE', kind='normal')
+    ch2_mass = Cpt(EpicsSignalRO, ':CH2_MASS', kind='normal')
+    ch3_size = Cpt(EpicsSignalRO, ':CH3_SIZE', kind='normal')
+    ch3_mass = Cpt(EpicsSignalRO, ':CH3_MASS', kind='normal')
+    ch4_size = Cpt(EpicsSignalRO, ':CH4_SIZE', kind='normal')
+    ch4_mass = Cpt(EpicsSignalRO, ':CH4_MASS', kind='normal')
+    ch5_size = Cpt(EpicsSignalRO, ':CH5_SIZE', kind='normal')
+    ch5_mass = Cpt(EpicsSignalRO, ':CH5_MASS', kind='normal')
+    Total_Mass = Cpt(EpicsSignalRO, ':TOT_MASS', kind='normal')
+    '''
+
+    # kind, doc are important keywords for later
+    # kind=hinted, normal, config, omitted
+
+
+
+ #Power Distribution Units: Temperature, Humidity, and Load Feed
+
+#PDU_Temp1: Will be for racks with 2 PDUs, with total of 3 sensors for monitoring purposes
+class PDU_Temp1(Device):
+    Sensor1 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[0]}:Sensor:1:GetTempValue', kind='hinted')
+    Sensor2 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[1]}:Sensor:2:GetTempValue', kind='hinted')
+    Sensor3 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[2]}:Sensor:1:GetTempValue', kind='hinted')
+    Sensor4 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[3]}:Sensor:2:GetTempValue', kind='hinted')
```

Comment:
Optional simplification: `FormattedComponent` is smart enough to assume self. So self doesn't need to be included in any of these. Optional of course, it works as-is so you don't have to change anything.

https://github.com/bluesky/ophyd/blob/f9be38c9b1de6338b4586b55eea55e74f3e18b2b/ophyd/device.py#L382-L383

Reply:
What does it mean for self to be assumed by FormattedComponent?

Reply:
It means you can leave it out and it knows to check `self.`. So, for example, Sensor4 here could have equivalently been:
```
Sensor4 = FCpt(EpicsSignalRO, '{prefix}:{elevation[3]}:Sensor:2:GetTempValue', kind='hinted')
```